src/key_events.cpp

```diff
@@ -40,57 +43,137 @@ static bool handleKeyswitchEventDefault(Key mappedKey, byte row, byte col, uint8
   return true;
 }
 
+void handleKeyswitchEvent(Key mappedKey, byte row, byte col, uint8_t keyState) {
+  if (!(keyState & INJECTED)) {
+    mappedKey = Layer.lookup(row, col);
+  }
+  for (byte i = 0; Kaleidoscope.eventHandlers[i] != NULL && i < HOOK_MAX; i++) {
+    Kaleidoscope_::eventHandlerHook handler = Kaleidoscope.eventHandlers[i];
+    mappedKey = (*handler)(mappedKey, row, col, keyState);
+    if (mappedKey.raw == Key_NoKey.raw)
+      return;
+  }
+  mappedKey = Layer.eventHandler(mappedKey, row, col, keyState);
+  if (mappedKey.raw == Key_NoKey.raw)
+    return;
+  handleKeyswitchEventDefault(mappedKey, row, col, keyState);
+}
+
+void pressKeyRaw(Key mappedKey) {
```

Comment:
This is one place where I'm a bit conflicted about naming. `pressKeyRaw` does not feel natural... `pressRawKey` or `pressKeyCode` might feel better, but I'm not sure about those, either. Mind you, `pressKeyRaw` does not feel wrong, either. Just... doesn't feel perfect, either.